pywinauto/controls/uia_controls.py

```diff
@@ -303,25 +303,37 @@ def select(self, item):
             # Try to access the underlying ListBox explicitly
             children_lst = self.children(control_type='List')
             if len(children_lst) > 0:
-                children_lst[0]._select(item)
+                list_view = children_lst[0]
+                list_view.get_item(item).select()
                 # do health check and apply workaround for Qt5 combo box if necessary
                 if isinstance(item, six.string_types):
-                    item = children_lst[0].children(name=item)[0]
-                    if self.selected_text() != item:
-                        # workaround for WinForms combo box
-                        item.invoke()
-                        if self.selected_text() != item:
-                            # workaround for Qt5 combo box
+                    item = list_view.children(name=item)[0]
+                    item_value = item.texts()[0]
```

Comment:
Why not getting `item.window_text()`? It gets only its own text instead of all children.